### PR TITLE
feat: allow filters to be disabled

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #6497 

### Description:

Allow for `FilterRule`s to be `disabled` by adding an optional prop `disabled`
This will allow users to interact with Dashboard Filters in this way: 
* They add a filter to the dashboard
* They don't know what value they want to filter by, _yet_

As proposed on the ticket, a `disabled`+clause that does no filtering (`1=1`) is being introduced here. However, not sure if there could be any issues with this approach for the current filters approach or if it should be done differently.

### CODE REVIEW

File: packages/common/src/types/filter.ts [hunk 1]
```diff
@@ -62,6 +62,7 @@ export type DashboardFilterRule<
 > = FilterRule<O, T, V, S> & {
     tileTargets?: Record<string, DashboardFieldTarget>;
     label: undefined | string;
+    disabled?: boolean;
```
Comment (owlas) [inline packages/common/src/types/filter.ts:65]: let's move to `filterRule`

File: packages/common/src/compiler/filtersCompiler.ts [hunk 2]
```diff
@@ -293,14 +294,18 @@ const renderBooleanFilterSql = (
...
+    filterRule: FilterRule<FilterOperator, unknown> | DashboardFilterRule,
     field: CompiledField,
     fieldQuoteChar: string,
     stringQuoteChar: string,
     escapeStringQuoteChar: string,
     startOfWeek: WeekDay | null | undefined,
     adapterType: SupportedDbtAdapter,
 ): string => {
+    if ('disabled' in filterRule && filterRule.disabled) {
+        return `WHERE 1=1`;
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/common/src/compiler/filtersCompiler.ts]: Got it! removed the `WHERE` and added a comment 👍

File: packages/common/src/types/filter.ts (hunk 1 above)
Comment (joaoviana) [reply packages/common/src/types/filter.ts]: done 👍

File: packages/common/src/compiler/filtersCompiler.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/common/src/compiler/filtersCompiler.ts]: should we be doing this or just returning undefined and then filtering undefined values when generating a final WHERE statement?

the question is: do we want disabled filters to show as 1=1 in the final query?

File: packages/common/src/compiler/filtersCompiler.ts [hunk 3]
```diff
@@ -301,6 +301,10 @@ export const renderFilterRuleSql = (
     startOfWeek: WeekDay | null | undefined,
     adapterType: SupportedDbtAdapter,
 ): string => {
+    if ('disabled' in filterRule && filterRule.disabled) {
```
Comment (owlas) [inline packages/common/src/compiler/filtersCompiler.ts:304]:
```suggestion
    if (!!filterRule.disabled) {
```

Comment (joaoviana) [reply packages/common/src/compiler/filtersCompiler.ts]: thank you 🙏 
Removed `!!` as ESLint told me it's redundant!

**Review by rephus — APPROVED** [review]
This looks good to me . I don't think this should break in any warehouse, right ? Recently @owlas tested something similar using `True` in all warehouses, not sure which one is better.

**Comment by ZeRego** [comment]
I think this might cause unexpected results because the selected fields and joined tables are still being affected by these disabled filters.
Check all the logic where we use `filters` in `buildQuery()` in `queryBuilder.ts`

**Review by ZeRego — CHANGES_REQUESTED** [review]
(details in comment)

**Comment by joaoviana** [comment]
putting in draft to address review comments + test this further

File: packages/backend/src/queryBuilder.test.ts [hunk 4]
```diff
@@ -133,6 +149,16 @@ describe('Query builder', () => {
...
+    test('Should build query with metric filter (where filter is disabled) and metric references a dimension from a joined table', () => {
+        expect(
+            buildQuery({
+                explore: EXPLORE,
+                compiledMetricQuery: METRIC_QUERY_WITH_METRIC_DISABLED_FILTER,
+                warehouseClient: warehouseClientMock,
+            }).query,
+        ).toStrictEqual(METRIC_QUERY_WITH_METRIC_FILTER_DISABLED_SQL);
+    });
+
```
Comment (joaoviana) [inline packages/backend/src/queryBuilder.test.ts:161]: @ZeRego could this match the scenario you were describing with metric filters + joined tables?

**Review by ZeRego — APPROVED** [review]
lgtm ⭐  thanks for the unit tests 😍

**Pull request merged**